phpBB/includes/ucp/ucp_pm.php

```diff
@@ -372,6 +372,14 @@ function main($id, $mode)
 					include($phpbb_root_path . 'includes/ucp/ucp_pm_viewfolder.' . $phpEx);
 					view_folder($id, $mode, $folder_id, $folder);
 
+					//After displaying messages on the left we set it up for composing a new message on the right
+					$action = 'post';
```

Comment:
Do not change action and mode here. Instead, use strings in `compose_pm` call like this:
```php
compose_pm($id, 'compose', 'post', $user_folders);
```